axelor-budget/src/main/java/com/axelor/apps/budget/service/GlobalBudgetServiceImpl.java

```diff
@@ -275,4 +278,31 @@ protected void recomputeImputedAmountsOnBudgetLine(BudgetLine budgetLine, Budget
     budgetLine.setFirmGap(budget.getTotalFirmGap());
     budgetLine.setAmountPaid(budget.getTotalAmountPaid());
   }
+
+  @Transactional
+  public void resetGlobalBudget(GlobalBudget globalBudget) {
+    globalBudget.setCode(globalBudget.getCode() + " (" + I18n.get("copy") + ")");
+    globalBudget.setStatusSelect(GlobalBudgetRepository.GLOBAL_BUDGET_STATUS_SELECT_DRAFT);
+    globalBudget.setArchived(false);
+
+    globalBudget.setTotalAmountCommitted(BigDecimal.ZERO);
+    globalBudget.setTotalAmountAvailable(globalBudget.getTotalAmountExpected());
+    globalBudget.setAvailableAmountWithSimulated(globalBudget.getTotalAmountExpected());
+    globalBudget.setRealizedWithNoPo(BigDecimal.ZERO);
+    globalBudget.setRealizedWithPo(BigDecimal.ZERO);
+    globalBudget.setSimulatedAmount(BigDecimal.ZERO);
+    globalBudget.setTotalFirmGap(BigDecimal.ZERO);
+    globalBudget.setTotalAmountPaid(BigDecimal.ZERO);
+
+    globalBudget.clearBudgetList();
+    List<BudgetLevel> budgetLevels = globalBudget.getBudgetLevelList();
+
+    if (ObjectUtils.notEmpty(budgetLevels)) {
+      budgetLevels.forEach(child -> budgetLevelService.resetBudgetLevel(child));
+    } else if (CollectionUtils.isNotEmpty(budgetLevels)) {
```

Comment:
You can remove the else if, can't be called and budgets will not be called from here